app/src/main/java/tech/ula/model/state/SessionStartupFsm.kt

```diff
@@ -198,79 +198,67 @@ class SessionStartupFsm(
         state.postValue(DownloadsHaveSucceeded)
     }
 
-    private suspend fun handleCopyDownloads(filesystem: Filesystem) = withContext(Dispatchers.IO) {
-        state.postValue(CopyingFilesToRequiredDirectories)
+    private suspend fun handleCopyDownloadsToLocalDirectories(filesystem: Filesystem) = withContext(Dispatchers.IO) {
+        state.postValue(CopyingFilesToLocalDirectories)
         try {
             downloadUtility.moveAssetsToCorrectLocalDirectory()
-            val copyingSucceeded = copyDistributionAssetsToFilesystem(filesystem)
-            if (!copyingSucceeded) {
-                state.postValue(DistributionCopyFailed)
-                return@withContext
-            }
             assetRepository.setLastDistributionUpdate(filesystem.distributionType)
         } catch (err: Exception) {
-            state.postValue(CopyingFailed)
+            state.postValue(LocalDirectoryCopyFailed)
             return@withContext
         }
-        state.postValue(CopyingSucceeded)
+        state.postValue(LocalDirectoryCopySucceeded)
+    }
+
+    private suspend fun handleVerifyFilesystemAssets(filesystem: Filesystem) = withContext(Dispatchers.IO) {
+        state.postValue(VerifyingFilesystemAssets)
+
+        val filesystemDirectoryName = "${filesystem.id}"
+        val requiredAssets = assetRepository.getDistributionAssetsForExistingFilesystem(filesystem)
+        val allAssetsArePresentOnFilesystem = filesystemUtility.areAllRequiredAssetsPresent(filesystemDirectoryName, requiredAssets)
+        val filesystemAssetsNeedUpdating = filesystem.lastUpdated < assetRepository.getLastDistributionUpdate(filesystem.distributionType)
+
+        if (!allAssetsArePresentOnFilesystem || filesystemAssetsNeedUpdating) {
+            if (!assetRepository.assetsArePresentInSupportDirectories(requiredAssets)) {
+                state.postValue(AssetsAreMissingFromSupportDirectories)
+                return@withContext
+            }
+
+            try {
+                filesystemUtility.copyAssetsToFilesystem("${filesystem.id}", filesystem.distributionType)
+                filesystem.lastUpdated = timeUtility.getCurrentTimeMillis()
+                filesystemDao.updateFilesystem(filesystem)
+            } catch (err: Exception) {
+                state.postValue(FilesystemAssetCopyFailed)
+            }
+
+            val filesystemDirectoryName = "${filesystem.id}"
```

Comment:
Does this variable need to be redeclared?  There is another variable with the same name in line 216

Reply:
It does not need to be redeclared, no.